modules/core/23-commitment/types/merkle.go

```diff
@@ -234,16 +226,8 @@ func (proof *MerkleProof) Empty() bool {
 	return proof == nil || proto.Equal(proof, blankMerkleProof) || proto.Equal(proof, blankProofOps)
 }
 
-// ValidateBasic checks if the proof is empty.
-func (proof MerkleProof) ValidateBasic() error {
-	if proof.Empty() {
-		return ErrInvalidProof
-	}
-	return nil
-}
-
 // validateVerificationArgs verifies the proof arguments are valid
-func (proof MerkleProof) validateVerificationArgs(specs []*ics23.ProofSpec, root exported.Root) error {
+func validateVerificationArgs(proof MerkleProof, path v2.MerklePath, specs []*ics23.ProofSpec, root exported.Root) error {
 	if proof.Empty() {
```

Comment:
maybe this should be `if proof == nil || proof.GetProofs() == nil`. I don't feel like the `proof.Empty()` is very effective and creates extra space in this file

Reply:
Support this 👍🏻

Reply:
Could also just call this `Validate(specs)` as method on merkle proof and move root nil checks to where its being used.

This is also totally fine, and happy to move along with it! 🙏🏻

Reply:
made the change, but left off the `Validate(specs)` suggestion. Happy to approve a followup, but wasn't exactly sure what changes you had in mind